src/lib/sstt/core/node.ml

```diff
@@ -188,11 +188,8 @@ include (struct
     let rec simplify t =
       if not t.simplified then begin
         let s_def = def t |> VDescr.simplify in
-        define ~simplified:true t s_def ;
-        s_def |> VDescr.direct_nodes |> List.iter simplify ;
-        match t.neg with
-        | None -> ()
-        | Some nt -> define ~simplified:true nt (VDescr.neg s_def)
+        define ~simplified:true t s_def;
+        s_def |> VDescr.direct_nodes |> List.iter simplify;
```

Comment:
The `match` was trying to exploit the fact that the negation of a simplified BDD is a simplified BDD, so that we can reuse the result of a simplification to simplify the negation too (the same fact is also exploited in `neg` by marking the newly created negation node with `~simplified:t.simplified`). Does it yield worse performance for some reasons?

Reply:
Oh actually dependencies are already invalidated when `define` is called.